dev/reference/tools/data-migration/upgrade.md

```diff
@@ -9,195 +9,131 @@ aliases: ['/docs/dev/how-to/upgrade/data-migration/']
-Git Commit Hash: cd753da958ea9a0d5686abc9f1988b61c9d36a89
-Git Branch: dm-master
-UTC Build Time: 2018-12-25 06:03:11
-Go Version: go version go1.11.2 linux/amd64
+Release Version: v1.0.0-10-geb2889c9
+Git Commit Hash: eb2889c9dcfbff6653be9c8720a32998b4627db9
+Git Branch: release-1.0
+UTC Build Time: 2019-09-06 03:18:48
+Go Version: go version go1.12 linux/amd64
 ```
 
 ### Main changes
 
-Before this version, DM-worker uses the following two ports to provide different information or services to the outside.
-
-- `dm_worker_port`: 10081 by default. Provides the RPC service to communicate with DM-master.
-- `dm_worker_status_port`: 10082 by default. Provides metrics and status information.
-
-Starting from this version, DM-worker uses a same port (8262 by default) to provide the above two kinds of information or services simultaneously.
+- Support automatically recovering replication tasks for some abnormal situations
+- Improve compatibility with DDL syntaxes
+- Fix the bug that the abnormal connection to the upstream database might cause data loss
 
 ### Upgrade operation example
 
-1. Modify the `inventory.ini` configuration information.
-
-    - Remove all the `dm_worker_status_port` configuration items and modify the `dm_worker_port` configuration item as needed.
-    - Remove all the `dm_master_status_port` configuration items and modify the `dm_master_port` configuration item as needed.
-
-    For example, modify
-
-    ```ini
-    dm_worker1_1 ansible_host=172.16.10.72 server_id=101 deploy_dir=/data1/dm_worker dm_worker_port=10081 dm_worker_status_port=10082 mysql_host=172.16.10.81 mysql_user=root mysql_password='VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU=' mysql_port=3306
-    ```
-
-    to
-
-    ```ini
-    dm_worker1_1 ansible_host=172.16.10.72 server_id=101 deploy_dir=/data1/dm_worker dm_worker_port=8262 mysql_host=172.16.10.81 mysql_user=root mysql_password='VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU=' mysql_port=3306
-    ```
-
-    and modify
-
-    ```ini
-    dm_master ansible_host=172.16.10.71 dm_master_port=12080 dm_master_status_port=12081
-    ```
-
-    to
-
-    ```ini
-    dm_master ansible_host=172.16.10.71 dm_master_port=8261
-    ```
+1. Download the new version of DM-Ansible, and confirm that there is `dm_version = v1.0.0` in the `inventory.ini` file.
+2. Run `ansible-playbook local_prepare.yml` to download the new DM binary file to the local disk.
+3. Run `ansible-playbook rolling_update.yml` to perform a rolling update for the DM cluster components.
+4. Run `ansible-playbook rolling_update_monitor.yml` to perform a rolling update for the DM monitoring components.
 
-2. Use DM-Ansible to perform a rolling update on DM, Prometheus and Grafana.
+> **Note:**
+>
+> When you upgrade DM to the 1.0 GA version, you must make sure that all DM cluster components (dmctl, DM-master, and DM-worker) are upgraded.
+>
+> For this particular upgrade operation, it is not supported to just upgrade a part of the components. Otherwise, an error might occur.
```

Comment:
When you upgrade DM to the 1.0 GA version, you must make sure that all DM cluster components (dmctl, DM-master, and DM-worker) are upgraded. Do not upgrade only a part of the components. Otherwise, an error might occur.